tutorials/ngsi_v2/e8_multientity_and_expression_language/e8_multientity_and_expression_language_solution.py

```diff
@@ -0,0 +1,166 @@
+                                 DeviceAttribute(name="fillingLevel", type="Number",
+                                                 expression="level / 100")
+                                 ]
+                     )
+    iota_client.post_device(device=device1)
+
+    # TODO: Setting expression language to JEXL at Device level with other attributes.
+    #  The attribute 'value' (Number) is itself multiplied by 5. The attribute
+    #  'consumption' (String) is the trimmed version of the attribute 'spaces' (String)
+    device2 = Device(device_id="waste_container_002",
+                     entity_name="urn:ngsi-ld:WasteContainer:002",
+                     entity_type="WasteContainer",
+                     transport=TransportProtocol.MQTT,
+                     protocol=PayloadProtocol.IOTA_JSON,
+                     expressionLanguage=ExpressionLanguage.JEXL,
+                     attributes=[DeviceAttribute(name="value", type="Number",
+                                                 expression="5 * value"),
+                                 DeviceAttribute(name="spaces", type="String"),
+                                 DeviceAttribute(name="consumption", type="String",
+                                                 expression="spaces | trim")
+                                 ]
+                     )
+    iota_client.post_device(device=device2)
+
+    client = mqtt_client.Client()
+    client.username_pw_set(username="", password="")
+    client.connect(MQTT_BROKER_HOST, MQTT_BROKER_PORT)
+    client.loop_start()
+
+    # TODO: Publish attributes 'level', 'longitude' and 'latitude' of device1
+    client.publish(topic=f'/json/{APIKEY}/{device1.device_id}/attrs',
+                   payload='{"level": 99, "longitude": 12.0, "latitude": 23.0}')
+
+    # TODO: Publish attributes 'value' and 'spaces' of device2
+    client.publish(topic=f'/json/{APIKEY}/{device2.device_id}/attrs',
+                   payload='{"value": 10, "spaces": "     foobar    "}')
+
+    client.disconnect()
+
+    # TODO: Create a weather station device with multi entity attributes (Number). 'v'
+    #  is multiplied by 100 and is a standard attribute. 'v1' and 'v2' are multiplied
+    #  by 100 and are multi entity attributes of type SubWeatherStation. The name of
+    #  each attribute is 'vol'.
+    device3 = Device(device_id="weather_station_001",
+                     entity_name="urn:ngsi-ld:WeatherStation:001",
+                     entity_type="WeatherStation",
+                     transport=TransportProtocol.MQTT,
+                     protocol=PayloadProtocol.IOTA_JSON,
+                     expressionLanguage=ExpressionLanguage.JEXL,
+                     attributes=[DeviceAttribute(object_id="v1", name="vol", type="Number",
+                                                 expression="v1*100",
+                                                 entity_name="urn:ngsi-ld:SubWeatherStation:001",
```

Comment:
In order to make this tutorial work for most FIWARE setups, e. g. regardless of the settings of IoT-Agent, I would suggest creating entities of `SubWeatherStation` before sending data. Otherwise, in some cases, there will be no `SubWeatherStation` entities

Reply:
@djs0109 The new commit creates the two entities of type `SubWeatherStation`. The IoT-Agent version 1.24.0 didn't work, but 1.26.0 worked for me.